mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -112,6 +119,63 @@ public CassandraMessageIdMapper(MailboxMapper mailboxMapper, CassandraMailboxDAO
             .sorted(Comparator.comparing(MailboxMessage::getUid));
     }
 
+    private CompletableFuture<Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>>>
+            retrieveMessages(FetchType fetchType, ImmutableList<ComposedMessageIdWithMetaData> composedMessageIds) {
+        return messageDAOV2.retrieveMessages(composedMessageIds, fetchType, UNLIMITED)
+                .thenCompose(messageResults -> FluentFutureStream.of(messageResults
```

Comment:
the code is really clutered by types and wrappers, could you try to enhance that a bit ?